File: src/Client/KeyboardModifiers.ts
/**
 * Enumerates modifier flags for keyboard accelerators.
 */
export const KeyboardModifiers = Object.freeze({

	/**
	 * Indicates no modifier.
	 */
	None: 0,

	/**
	 * Indicates the `Control` modifier.
	 */
	Ctrl: 1,

	/**
	 * Indicates the `Shift` modifier.
	 */
	Shift: 2,

	/**
	 * Indicates the `Alt` modifier (`Option` key on macOS, `Menu` key on Windows).
	 */
	Alt: 4,

	/**
	 * Indicates the `Meta` modifier (`Command` key on macOS, `Windows` key on Windows).
	 */
	Meta: 8
});

File: Sources/Server/KeyboardModifiers.cs
namespace Belin.UI;

/// <summary>
/// Enumerates modifier flags for keyboard accelerators.
/// </summary>
[Flags]
public enum KeyboardModifiers {

	/// <summary>
	/// Indicates no modifier.
	/// </summary>
	None,

	/// <summary>
	/// Indicates the <c>Control</c> modifier.
	/// </summary>
	Ctrl = 1,

	/// <summary>
	/// Indicates the <c>Shift</c> modifier.
	/// </summary>
	Shift = 2,

	/// <summary>
	/// Indicates the <c>Alt</c> modifier (<c>Option</c> key on macOS, <c>Menu</c> key on Windows).
	/// </summary>
	Alt = 4,

	/// <summary>
	/// Indicates the <c>Meta</c> modifier (<c>Command</c> key on macOS, <c>Windows</c> key on Windows).
	/// </summary>
	Meta = 8
}
